src/liblzma/check/crc32_loongarch.h

```diff
@@ -0,0 +1,107 @@
+// SPDX-License-Identifier: 0BSD
+
+///////////////////////////////////////////////////////////////////////////////
+//
+/// \file       crc32_loongarch.h
+/// \brief      CRC32 calculation with LoongArch optimization
+//
+//  Authors:    Xi Ruoyao
+//
+///////////////////////////////////////////////////////////////////////////////
+
+
+#ifndef LZMA_CRC32_LOONGARCH_H
+#define LZMA_CRC32_LOONGARCH_H
+
+#include <larchintrin.h>
+#include <sys/auxv.h>
+
+// Defined in section 9.1 of
+// "Software Development and Build Convention for LoongArch Architectures"
+// (https://github.com/loongson/la-softdev-convention/).  Hard code it
+// instead of including <bits/hwcap.h> for compatibility with Glibc <= 2.37.
+#define HWCAP_LOONGARCH_CRC32 (1 << 6)
+
+#if __loongarch_grlen >= 64
+#  define UNIT_SIZE 8
+#  define CRC_UNIT(buf, crc) __crc_w_d_w((int64_t)aligned_read64le(buf), crc)
+#else
+#  define UNIT_SIZE 4
+#  define CRC_UNIT(buf, crc) __crc_w_w_w((int32_t)aligned_read32le(buf), crc)
+#endif
+
+static uint32_t
+crc32_arch_optimized(const uint8_t *buf, size_t size, uint32_t _crc)
+{
+	int32_t crc = (int32_t)~_crc;
+
+	// Align the input buffer because this was shown to be
+	// significantly faster than unaligned accesses.
+	const size_t align_amount = my_min(size,
+			(UNIT_SIZE - (uintptr_t)buf) & (UNIT_SIZE - 1));
+
+	if (align_amount & 1)
+		crc = __crc_w_b_w((int8_t)*buf++, crc);
+
+	if (align_amount & 2) {
+		crc = __crc_w_h_w((int16_t)aligned_read16le(buf), crc);
+		buf += 2;
+	}
+
+#if UNIT_SIZE == 8
+	if (align_amount & 4) {
+		crc = __crc_w_w_w((int32_t)aligned_read32le(buf), crc);
+		buf += 4;
+	}
+#endif
+
+	size -= align_amount;
+
+	// Process UNIT_SIZE bytes at a time. The end point is determined by
+	// ignoring the least significant log2(UNIT_SIZE) bits of size to ensure
+	// we do not process past the bounds of the buffer. This guarantees
+	// that limit is a multiple of UNIT_SIZE and is strictly less than size.
+	for (const uint8_t *limit = buf + (size & ~((size_t)UNIT_SIZE - 1));
+			buf < limit; buf += UNIT_SIZE)
+		crc = CRC_UNIT(buf, crc);
+
+	// Process the remaining bytes that are not 8 byte aligned.
```

Comment:
This comment needs updating to reference `UNIT_SIZE` instead of "8".

Reply:
Done.